src/SonarScanner.MSBuild.PreProcessor/ProcessedArgs.cs

```diff
@@ -286,6 +295,45 @@ ServerInfo Warn(ServerInfo server, string message)
                 return server;
             }
         }
+
+        private bool TryGetUserHome(ILogger logger, IDirectoryWrapper directoryWrapper, IOperatingSystemProvider operatingSystemProvider, out string userHome)
```

Comment:
`operatingSystemProvider` is a field already. It does not need to be passed as a parameter. We might want to do the same with the logger and directoryWrapper.

```suggestion
        private bool TryGetUserHome(ILogger logger, IDirectoryWrapper directoryWrapper, out string userHome)
```

Reply:
Fair point, but this requires a complete rewrite of the class. I wouldn't do this as part of this PR.
Another thought: We have the `ProcessedArgs` class and the `ArgumentProcessor`. Maybe we should turn ProcessedArgs into a dumb POCO and move all the logic and validation to ArgumentProcessor.
It seems very unfortunate that the constructor of ProcessedArgs does validation and has so many dependencies while its main purpose is to hold a bunch of validated arguments.
If we move to ArgumentProcessor, we could also solve the problem of passing some of the wrappers and the logger while others are stored as fields.

Reply:
You can just remove the parameter for now. No other changes are needed in this PR.

And I also like your idea, let's do this at the end of the sprint if we have some time.